snap/info.go

```diff
@@ -208,6 +208,17 @@ func ScopedSecurityTag(snapName, scopeName, suffix string) string {
 	return fmt.Sprintf("snap.%s.%s.%s", snapName, scopeName, suffix)
 }
 
+// SplitSecurityTag will extract the snap name from a string of the form
+// `snap.<snap>[.<app>]`. If the tag is not of that form, returns an error.
```

Comment:
I think we have more forms now because of hooks, this probably need test where is given as input the result of any of *SecurityTag functions in this package. There might be more coming as well because of components